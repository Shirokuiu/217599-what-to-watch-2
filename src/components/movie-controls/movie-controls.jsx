import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";

import {ActionCreator} from "../../reducer";

export const MovieControls = (props) => {
  const ParentStatus = {
    PROMO: `promo`,
    MOVIE_PAGE: `moviePage`
  };

  const {parent, state, onButtonClick, movieId, onPlayClick} = props;
  const {isButtonDisable, isFavorite} = state;

  const handlePlayClick = () => {
    onPlayClick(movieId);
  };

  return <div className="movie-card__buttons">
    <Link to={`/movie/${movieId}/watch`} onClick={handlePlayClick} className="btn btn--play movie-card__button">
      <svg viewBox="0 0 19 19" width="19" height="19">
        <use xlinkHref="#play-s"/>
      </svg>
      <span>Play</span>
    </Link>
    <button
      className="btn btn--list movie-card__button"
      id="myList"
      type="button"
      disabled={isButtonDisable}
      onClick={onButtonClick}
    >
      {isFavorite ?
        <svg viewBox="0 0 18 14" width="18" height="14">
          <use xlinkHref="#in-list"/>
        </svg> :
        <svg viewBox="0 0 19 20" width="19" height="20">
          <use xlinkHref="#add"/>
        </svg>}
      <span>My list</span>
    </button>
    {ParentStatus[parent] === ParentStatus.MOVIE_PAGE ?
      <a href="add-review.html" id="addReview" className="btn movie-card__button" onClick={onButtonClick}>Add review</a> :
      null
    }
  </div>;
};

MovieControls.propTypes = {
  parent: PropTypes.string.isRequired,
  state: PropTypes.shape({
    isFavorite: PropTypes.bool,
    isButtonDisable: PropTypes.bool
  }),
  onButtonClick: PropTypes.func,
  movieId: PropTypes.number,
  onPlayClick: PropTypes.func
};

const mapDispatchToProps = (dispatch) => ({
  onPlayClick: (movieId) => {
    dispatch(ActionCreator.updateCurrentMovie(movieId));
  }
});

export default connect(null, mapDispatchToProps)(MovieControls);
