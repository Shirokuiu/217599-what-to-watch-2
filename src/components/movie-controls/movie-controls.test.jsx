import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";

import {MovieControls} from "../movie-controls/movie-controls";

it(`Is movie controls rendered`, () => {
  const tree = renderer.create(<BrowserRouter>
    <MovieControls
      state={{
        isFavorite: true,
        isButtonDisable: false
      }}
      onButtonClick={jest.fn()}
      parent={`promo`}
    />
  </BrowserRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
