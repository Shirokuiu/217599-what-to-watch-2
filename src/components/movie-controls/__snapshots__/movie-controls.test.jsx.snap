// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is movie controls rendered 1`] = `
<div
  className="movie-card__buttons"
>
  <a
    className="btn btn--play movie-card__button"
    href="/movie/undefined/watch"
    onClick={[Function]}
  >
    <svg
      height="19"
      viewBox="0 0 19 19"
      width="19"
    >
      <use
        xlinkHref="#play-s"
      />
    </svg>
    <span>
      Play
    </span>
  </a>
  <button
    className="btn btn--list movie-card__button"
    disabled={false}
    id="myList"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      height="14"
      viewBox="0 0 18 14"
      width="18"
    >
      <use
        xlinkHref="#in-list"
      />
    </svg>
    <span>
      My list
    </span>
  </button>
</div>
`;
